core/l10n_init.js

```diff
@@ -115,6 +129,7 @@
         getLocale: getLocale,
         getLocaleSrc: getLocaleSrc,
         injectLocaleScript: injectLocaleScript,
-        languages: languages
+        languages: languages,
+        promise: injectLocaleScript().promise
```

Comment:
I'd prefer that we call `injectLocaleScript` in `core/init` rather than kicking it off here?